src/proof/mod.rs

```diff
@@ -215,4 +215,31 @@ pub trait Prover<F: CurveCycleEquipped> {
         let unfull_multiframe_frame_count = raw_iterations % rc;
         full_multiframe_count + usize::from(unfull_multiframe_frame_count != 0)
     }
+
+    /// Generate a proof from a sequence of frames
+    fn prove_from_frames(
+        &self,
+        pp: &Self::PublicParams,
+        frames: &[crate::lem::interpreter::Frame],
+        store: &Arc<Store<F>>,
+        init: Option<
+            <Self::RecursiveSNARK as RecursiveSNARKTrait<F, Self::Frame>>::BaseRecursiveSNARK,
+        >,
+    ) -> Result<(Self::RecursiveSNARK, Vec<F>, Vec<F>, usize), ProofError> {
+        let folding_config = self
+            .folding_mode()
+            .folding_config(self.lang().clone(), self.reduction_count());
+        let steps = Self::from_frames(frames, store, &folding_config.into());
+        self.prove(pp, steps, store, init)
+    }
+
+    /// Returns the `Lang` wrapped with `Arc` for cheap cloning
+    fn lang(&self) -> &Arc<Lang<F, C>>;
+
+    /// Converts input into Self::Frames according to the rules of the Prover
+    fn from_frames(
```

Comment:
I wasn't thrilled about having to add this method, but I didn't have a better way. Suggestions welcome!

Reply:
Can you provide the default implementation in the trait itself so we don't need to replicate them on both provers?

Reply:
Well, the problem is that `Frame` is an associated type of the `Prover` trait, so I can't make it specific to `MultiFrame` (or any other concrete type). I could make it generic over `Frame`, but that seems bad to me.